final_task/rss_reader/news.py

```diff
@@ -0,0 +1,141 @@
+
+                if flag:
+                    print(line, end='')
+                    empty = False
+
+                if '!!!#####STARTOFNEWS#####!!!'+date in line:
+                    flag = True
+
+                    if counter == limit:
+                        return 3
+
+                    if limit != -1:
+                        counter += 1
+    except FileNotFoundError:
+        print('Error: File not found. (Maybe this is a first time you are running a program)')
+        return 1
+
+    if empty:
+        print('Error: news with ' + date + ' not found.')
+        return 2
+
+    return 0
+
+
+def news_decompose(date: str, limit: int):
+    """This function takes the news from local storage"""
+    flag = False
+    empty = True
+    counter_news = 0
+    counter_lines = 0
+    counter = 0
+    news = dict()
+    dict_buffer = dict()
+    news_buffer = list()
+    string_buffer = ''
+
+    try:
+        with open(directory, 'r', encoding='utf-8') as file:
+            for line in file:
+                if '!!!#####ENDOFNEWS#####!!!' + date in line:
+                    flag = False
+
+                    if string_buffer:
+                        dict_buffer['description'] = string_buffer
+                        news[counter_news] = dict_buffer
+                        counter_news += 1
+
+                    counter_lines = 0
+                    news_buffer = list()
+                    dict_buffer = dict()
+                    string_buffer = ''
+
+                if flag:
+                    empty = False
+                    if counter_lines < 5:
+                        news_buffer.append(line)
+                        counter_lines += 1
+                    elif counter_lines == 5:
+                        dict_buffer['feed'] = news_buffer[0].replace('Feed: ', '').replace('\n', '')
+                        dict_buffer['title'] = news_buffer[1].replace('Title: ', '').replace('\n', '')
+                        dict_buffer['date'] = news_buffer[2].replace('Date: ', '').replace('\n', '')
+                        dict_buffer['link'] = news_buffer[3].replace('Link: ', '').replace('\n', '')
+                        dict_buffer['image'] = news_buffer[4].replace('Image: ', '').replace('\n', '')
+                        counter_lines += 1
+                    else:
+                        string_buffer += line
+
+                if '!!!#####STARTOFNEWS#####!!!' + date in line:
+                    flag = True
+
+                    if counter == limit:
+                        return news
+
+                    if limit != -1:
+                        counter += 1
+
+        if empty:
+            print('Error: news with ' + date + ' not found.')
+            news[0] = 2
+            return news
+
+        return news
+
+    except FileNotFoundError:
+        print('Error: File not found. (Maybe this is a first time you are running a program)')
+        news[0] = 1
+        return news
+
+
+
+
+
+
+
+
```

Comment:
Please leave only 1 empty line at the end of a file

Reply:
Fixed.